core/src/unprocessed_packet_batches.rs

```diff
@@ -79,6 +90,50 @@ impl UnprocessedPacketBatches {
     pub fn with_capacity(capacity: usize) -> Self {
         UnprocessedPacketBatches(VecDeque::with_capacity(capacity))
     }
+
+    /// Returns total number of all packets (including unprocessed and processed) in buffer
+    #[allow(dead_code)]
+    fn get_packets_count(&self) -> usize {
+        self.iter()
+            .map(|deserialized_packet_batch| deserialized_packet_batch.packet_batch.packets.len())
+            .sum()
+    }
+
+    /// Returns total number of unprocessed packets in buffer
+    #[allow(dead_code)]
+    fn get_unprocessed_packets_count(&self) -> usize {
+        self.iter()
+            .map(|deserialized_packet_batch| deserialized_packet_batch.unprocessed_packets.len())
+            .sum()
+    }
+
+    /// Iterates packets in buffered packet_batches, returns all unprocessed packet's stake,
+    /// and its locator
+    #[allow(dead_code)]
+    fn get_stakes_and_locators(&self) -> (Vec<u64>, Vec<PacketLocator>) {
```

Comment:
This returns direct pairings, right? Maybe `Vec<(u64, PacketLocator)>` instead?

Reply:
Mmm, stakes and locators will be used separately, caller would have to `unzip` if returns a vector of tuples.

Reply:
Oh.  Can you add a test (or modify the current one) to demonstrate that independent usage?  It's not intuitive to me how they are useful apart from each other

Reply:
example is here I think: https://github.com/solana-labs/solana/blob/7aba3ad2afbd92a2321d9eac9d670a72faba42ca/core/src/unprocessed_packet_batches.rs#L165-L169 from the big PR: https://github.com/solana-labs/solana/pull/23257/files

Reply:
Yep, it comes in next PR 😄 
Basically `stakes` is used once as input to weight shuffle, but the `locators` will be used in the remaining process.

Reply:
got it, thanks!